flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/StopWithSavepointTest.java

```diff
@@ -377,6 +378,7 @@ void testOnGloballyTerminalStateWaitsForSavepointCompletion() throws Exception {
             ctx.setStopWithSavepoint(sws);
 
             sws.onGloballyTerminalState(JobStatus.FINISHED);
+            // this is a sanity check that we haven't scheduled a state transition
```

Comment:
there's another location in `testConcurrentSavepointFailureAndGloballyTerminalStateCauseRestart` (see [line 407](https://github.com/apache/flink/blob/106b83b88a1dab577b2f60b5a9dc8052526d3820/flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/StopWithSavepointTest.java#L407)).